<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_window</class>
 <widget class="QMainWindow" name="main_window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>665</width>
    <height>751</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>通信代码生成工具</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="load_xml">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>10</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>加载</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_xml">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>10</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>保存</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="xml_path">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>10</y>
      <width>291</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>xml保存的全路径，可以通过浏览按钮获取</string>
    </property>
   </widget>
   <widget class="QPushButton" name="view_bt">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>10</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>浏览</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>63</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>xml路径</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="project">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>50</y>
      <width>81</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>项目名称</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>63</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>projcet</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>40</y>
      <width>63</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>port</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>282</x>
      <y>40</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>client type</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_cli">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>50</y>
      <width>79</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="port_box">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>50</y>
      <width>61</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>65536</number>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_serv">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>50</y>
      <width>79</width>
      <height>28</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>40</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>server type</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>client namespace</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cli_namespace">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>90</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>必填项</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>80</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>server namespace</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="serv_namespace">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>90</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>必填项</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="serv_classname">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>130</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>默认与namespace相同</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>120</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>client classname</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cli_classname">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>130</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>默认与namespace相同</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>120</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>server classname</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>160</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>client filename</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cli_file">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>170</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>必填项</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="serv_file">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>170</y>
      <width>181</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>必填项</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>160</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>server filename</string>
    </property>
   </widget>
   <widget class="QPushButton" name="add_func">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>添加函数</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rm_func">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>210</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>删除函数</string>
    </property>
   </widget>
   <widget class="QPushButton" name="edit_bt">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>210</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>编辑函数</string>
    </property>
   </widget>
   <widget class="QPushButton" name="view_bt_cli">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>630</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>浏览</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>630</y>
      <width>71</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>client路径</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="cli_path">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>630</y>
      <width>291</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>xml保存的全路径，可以通过浏览按钮获取</string>
    </property>
   </widget>
   <widget class="QPushButton" name="view_bt_server">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>670</y>
      <width>84</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>浏览</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>670</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>server路径</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="server_path">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>670</y>
      <width>291</width>
      <height>28</height>
     </rect>
    </property>
    <property name="placeholderText">
     <string>xml保存的全路径，可以通过浏览按钮获取</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generate">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>620</y>
      <width>141</width>
      <height>81</height>
     </rect>
    </property>
    <property name="text">
     <string>开始生成</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="withping">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>210</y>
      <width>191</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>client是否开启保活服务</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="func_list">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>250</y>
      <width>621</width>
      <height>359</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="mouseTracking">
     <bool>true</bool>
    </property>
    <property name="tabletTracking">
     <bool>true</bool>
    </property>
    <property name="focusPolicy">
     <enum>Qt::ClickFocus</enum>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>20</number>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAsNeeded</enum>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QAbstractScrollArea::AdjustToContents</enum>
    </property>
    <property name="autoScroll">
     <bool>true</bool>
    </property>
    <property name="tabKeyNavigation">
     <bool>false</bool>
    </property>
    <property name="showDropIndicator" stdset="0">
     <bool>false</bool>
    </property>
    <property name="dragDropOverwriteMode">
     <bool>false</bool>
    </property>
    <property name="alternatingRowColors">
     <bool>true</bool>
    </property>
    <property name="selectionMode">
     <enum>QAbstractItemView::SingleSelection</enum>
    </property>
    <property name="selectionBehavior">
     <enum>QAbstractItemView::SelectRows</enum>
    </property>
    <property name="verticalScrollMode">
     <enum>QAbstractItemView::ScrollPerPixel</enum>
    </property>
    <property name="horizontalScrollMode">
     <enum>QAbstractItemView::ScrollPerPixel</enum>
    </property>
    <property name="gridStyle">
     <enum>Qt::CustomDashLine</enum>
    </property>
    <property name="columnCount">
     <number>1</number>
    </property>
    <attribute name="horizontalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>619</number>
    </attribute>
    <attribute name="horizontalHeaderMinimumSectionSize">
     <number>50</number>
    </attribute>
    <attribute name="verticalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <attribute name="verticalHeaderHighlightSections">
     <bool>false</bool>
    </attribute>
    <attribute name="verticalHeaderShowSortIndicator" stdset="0">
     <bool>true</bool>
    </attribute>
    <attribute name="verticalHeaderStretchLastSection">
     <bool>true</bool>
    </attribute>
    <column>
     <property name="text">
      <string>函数的逻辑表达式</string>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
    </column>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
